modules/ppcp-wc-gateway/src/Helper/PWCProductStatus.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * Status of the Pay With Crypto merchant connection.
+ *
+ * @package WooCommerce\PayPalCommerce\WcGateway\Helper
+ */
+
+declare(strict_types=1);
+
+namespace WooCommerce\PayPalCommerce\WcGateway\Helper;
+
+use WooCommerce\PayPalCommerce\ApiClient\Endpoint\PartnersEndpoint;
+use WooCommerce\PayPalCommerce\ApiClient\Entity\SellerStatusCapability;
+use WooCommerce\PayPalCommerce\ApiClient\Helper\Cache;
+use WooCommerce\PayPalCommerce\ApiClient\Helper\FailureRegistry;
+use WooCommerce\PayPalCommerce\WcGateway\Settings\Settings;
+use WooCommerce\PayPalCommerce\ApiClient\Helper\ProductStatus;
+use WooCommerce\PayPalCommerce\ApiClient\Entity\SellerStatus;
+
+/**
+ * Class PWCProductStatus
+ */
+class PWCProductStatus extends ProductStatus {
+	public const CAPABILITY_NAME      = 'CRYPTO_PYMTS';
+	public const SETTINGS_KEY         = 'products_pwc_enabled';
+	public const PWC_STATUS_CACHE_KEY = 'pwc_status_cache';
+
+	public const SETTINGS_VALUE_ENABLED   = 'yes';
+	public const SETTINGS_VALUE_DISABLED  = 'no';
+	public const SETTINGS_VALUE_UNDEFINED = '';
+
+	/**
+	 * The Cache.
+	 *
+	 * @var Cache
+	 */
+	protected Cache $cache;
+
+	/**
+	 * The settings.
+	 *
+	 * @var Settings
+	 */
+	private Settings $settings;
+
+	/**
+	 * PWCProductStatus constructor.
+	 *
+	 * @param Settings         $settings             The Settings.
+	 * @param PartnersEndpoint $partners_endpoint    The Partner Endpoint.
+	 * @param Cache            $cache                The cache.
+	 * @param bool             $is_connected         The onboarding state.
+	 * @param FailureRegistry  $api_failure_registry The API failure registry.
+	 */
+	public function __construct(
+		Settings $settings,
+		PartnersEndpoint $partners_endpoint,
+		Cache $cache,
+		bool $is_connected,
+		FailureRegistry $api_failure_registry
+	) {
+		parent::__construct( $is_connected, $partners_endpoint, $api_failure_registry );
+
+		$this->settings = $settings;
+		$this->cache    = $cache;
+	}
+
+	/** {@inheritDoc} */
```

Comment:
Also the inheritDoc comments are not needed, If I am not wrong the CS will not complain and the modern IDEs automatically understand these ones.